web/bfex/components/data_pipeline/tasks/scrape.py

```diff
@@ -32,16 +33,39 @@ def run(self, data):
         :param data: str or Faculty instance.
         :return: tuple of the faculty name and Scrapp produced by scraping the faculty directory page.
         """
+        if isinstance(data, str):
+            faculty_name = data
+        else:
+            faculty_name = data.name
+
+        search_results = Faculty.search().query('match', name=faculty_name).execute()
+        if len(search_results) > 1:
+            # Shouldn't happen, but could.
+            raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
 
-        faculty_directory_url = URLs.build_faculty_url(data)
+        faculty = search_results[0]
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="FacultyPage")
+        search_dup.delete()
+            
+        faculty_directory_url = URLs.build_faculty_url(faculty_name)
 
         scraper = ScraperFactory.create_scraper(faculty_directory_url, ScraperType.PROFILE)
         scrapp = scraper.get_scrapps()[0]
 
+        doc = Document()
+        doc.source = "FacultyPage"
```

Comment:
Weren't we calling this profile before? Has it been adjusted wherever else it was being used?

Reply:
I thought @morlick was calling profile the thing from research id, let me invesgigate

Reply:
ugh, no garry I am dumb

Reply:
omg you already coded these 4 lines, deleted now. I am so dumb